packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -300,17 +301,18 @@ export class SchemaFactory<
 			public constructor(input: FlexTreeNode | unknown) {
 				super();
 				// Currently this just does validation. All other logic is in the subclass.
-				if (isFlexTreeNode(input)) {
-					assert(
-						getClassSchema(input.schema) === this.constructor,
-						0x83b /* building node with wrong schema */,
+				if (isFlexTreeNode(input) && getClassSchema(input.schema) !== this.constructor) {
+					throw new TypeError(
+						`The provided input is incompatible with "${schema.identifier}" schema.`,
+					);
+				}
+
+				if (isTreeNode(input)) {
+					// TODO: update this once we have better support for deep-copying and move operations.
+					throw new UsageError(
+						"Existing nodes cannot be used as new content to insert into the tree. Create a new node using your schema factory instead.",
```

Comment:
I'm not sure that's a good description. If someone does `new Foo(new Foo({}))` (which will trigger this) when they wanted `new Foo({child: new Foo({})}) I'm not sure this is accurate.

Also if they did `const x = new Foo(foo)` when they want `const x = foo` (this is the case that I saw which motivated this) I think its also not useful.

Additionally, a SchemaFactory is a thing, and it is not involved in making node, but instead makes schema which isn't relevant here.

```suggestion
						"Existing nodes cannot be used as the constructor parameter for a new node. The existing node may be able to be used directly instead of creating a new one, used as a child of the new node (if its Unhydrated). If the desired result is copying the provided node, it must be deep copied (since child node can be parented under both the new and old one). Currently no API is provided to make deep copies, but it can be done manually with object spreads, for example `new Foo({...oldFoo})` will work if all fields of foo are leaf nodes",
```

Reply:
Since we presumably don't expose the concept of hydration to users, I took a stab at adapting your recommendation to avoid referencing internal concepts. Let me know what you think.